perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -41,6 +43,405 @@
+    fr_UDP500_name = ('fr-udp500-%s-%s' %
+                      (suffix, FLAGS.run_uri))
+    fr_UDP4500_name = ('fr-udp4500-%s-%s' %
+                       (suffix, FLAGS.run_uri))
+    fr_ESP_name = ('fr-esp-%s-%s' %
+                   (suffix, FLAGS.run_uri))
+
+    if fr_UDP500_name not in self.forwarding_rules:
+      fr_UDP500 = GceForwardingRule(
+          fr_UDP500_name, 'UDP', self, 500)
+      self.forwarding_rules[fr_UDP500_name] = fr_UDP500
+      fr_UDP500.Create()
+    if fr_UDP4500_name not in self.forwarding_rules:
+      fr_UDP4500 = GceForwardingRule(
+          fr_UDP4500_name, 'UDP', self, 4500)
+      self.forwarding_rules[fr_UDP4500_name] = fr_UDP4500
+      fr_UDP4500.Create()
+    if fr_ESP_name not in self.forwarding_rules:
+      fr_ESP = GceForwardingRule(
+          fr_ESP_name, 'ESP', self)
+      self.forwarding_rules[fr_ESP_name] = fr_ESP
+      fr_ESP.Create()
+
+  def _SetupRouting(self, suffix, next_hop_tun, dest_cidr):
+    """Create IPSec routing rules between the source gw and the target gw.
+
+    Args:
+      target_gw: The VPNGW object to point routing rules at.
+    """
+
+    route_name = 'route-' + self.name + '-' + suffix
+    if route_name not in self.routes:
+      self.routes[route_name] = GceRoute(route_name, dest_cidr, self.network_name, next_hop_tun, self.region, self.project, self.region)
+      self.routes[route_name].Create()
+
+  def Create(self):
+    """Creates the actual VPNGW."""
+    benchmark_spec = context.GetThreadBenchmarkSpec()
+    if benchmark_spec is None:
+      raise errors.Error('GetNetwork called in a thread without a '
+                         'BenchmarkSpec.')
+    if self.created:
+      return
+    if self.vpn_gw_resource:
+      self.vpn_gw_resource.Create()
+    key = self.name
+    if key not in benchmark_spec.vpn_gws:
+      benchmark_spec.vpn_gws[key] = self
+
+    self.created = True
+    return benchmark_spec.vpn_gws[key]
+
+  def Delete(self):
+    """Deletes the actual VPNGW."""
+    if self.ip_resource:
+      self.ip_resource.Delete()
+
+    if self.tunnels:
+      for tun in self.tunnels:
+        self.tunnels[tun].Delete()
+
+    if self.forwarding_rules:
+      for fr in self.forwarding_rules:
+        self.forwarding_rules[fr].Delete()
+
+    if self.routes:
+      for route in self.routes:
+        self.routes[route].Delete()
+
+    if self.vpn_gw_resource:
+      self.vpn_gw_resource.Delete()
+
+    self.created = False
+
+
+class GceVPNGWResource(resource.BaseResource):
+
+  def __init__(self, name, network_name, region, cidr, project):
+    super(GceVPNGWResource, self).__init__()
+    self.name = name
+    self.network_name = network_name
+    self.region = region
+    self.cidr = cidr
+    self.project = project
+
+  def _Create(self):
+    cmd = util.GcloudCommand(self, 'compute', 'target-vpn-gateways', 'create',
+                             self.name)
+    cmd.flags['network'] = self.network_name
+    cmd.flags['region'] = self.region
```

Comment:
Since cmd.flag['region'] is used frequently, consider moving it to providers/gcp/util.py - inside _AddCommonFlags

Reply:
Fixed.